### PR TITLE
Pfm 5103 ifms adapter sanction and allocation

### PR DESCRIPTION
**Pull request by ansh-egov**

### CODE REVIEW

File: reference-adapters/ifms-adapter/src/main/java/org/egov/service/DisbursementService.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        PISearchCriteria piSearchCriteria = PISearchCriteria.builder()
+                .muktaReferenceId(disbursementRequest.getMessage().getTargetId())
+                .tenantId(disbursementRequest.getMessage().getLocationCode())
+                .sortBy(PISearchCriteria.SortBy.createdTime)
+                .sortOrder(PISearchCriteria.SortOrder.DESC)
+                .isActive(true)
+                .build();
+        List<PaymentInstruction> paymentInstructions = piRepository.searchPi(piSearchCriteria);
+        SanctionDetailsSearchCriteria sanctionDetailsSearchCriteria = SanctionDetailsSearchCriteria.builder()
+                .ids(Collections.singletonList(disbursementRequest.getMessage().getSanctionId()))
```
[the remark this answers is not in the document]
Comment (ansh-egov) [reply reference-adapters/ifms-adapter/src/main/java/org/egov/service/DisbursementService.java:76]: its already there in SanctionDetailsSearchCriteria

File: reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/Type.java
Comment (shailesh-egov) [inline reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/Type.java:1]: @ansh-egov what is this type ?

File: reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/RequestStatus.java
Comment (shailesh-egov) [inline reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/RequestStatus.java:1]: @ansh-egov are we using this?

File: reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/MsgHeaderStatusReasonCode.java
Comment (shailesh-egov) [inline reference-adapters/ifms-adapter/src/main/java/org/egov/web/models/enums/MsgHeaderStatusReasonCode.java:1]: @ansh-egov are we using this?

**Review by shailesh-egov — COMMENTED** [review]
Remove un-used files.

File: reference-adapters/ifms-adapter/src/main/java/org/egov/service/FailureDetailsService.java [hunk 2]
```diff
@@ -172,48 +152,51 @@ private void addReversalTransactionAndUpdatePIPa(PaymentInstruction pi, Payment
...
             piUtils.updatePIIndex(requestInfo, pi);
+            //Call On disburse for the PI
+            paymentInstructionService.processPIForOnDisburse(pi, requestInfo);
 
         } catch (Exception e) {
             log.error("Failed in FailureDetailsService:addReversalTransactionAndUpdatePIPa " + e);
         }
     }
 
-    private void updatePaymentStatusForPartial(Payment payment, RequestInfo requestInfo) {
```
Comment (shailesh-egov) [inline reference-adapters/ifms-adapter/src/main/java/org/egov/service/FailureDetailsService.java:184]: @ansh-egov use multi-line comment

**Review by shailesh-egov — COMMENTED** [review]
@ansh-egov please add logs and comments.

**Review by shailesh-egov — APPROVED** [review]
Remove testing code once enviornment is ready.

**Pull request merged**